apps/frontend/core-dapp/src/stores/graphs/CoreGraphStoreV2.ts

```diff
@@ -0,0 +1,85 @@
+import { GRAPH_ENDPOINTS } from 'prepo-constants'
+import { makeAutoObservable } from 'mobx'
+import { createHttpClient } from 'mst-gql'
+import { getContractAddress } from 'prepo-utils'
+import { RootStore } from '../RootStore'
+import {
+  KNOWN_HISTORY_EVENTS,
+  KNOWN_HISTORY_EVENTS_MAP,
+} from '../../features/history/history-utils'
+import { RootStoreBase } from '../../../generated/mst-gql/core-dapp/RootStore.base'
+import { FilterType } from '../../components/Filter/filter.constants'
+import { supportedContracts } from '../../lib/supported-contracts'
+import { SubgraphErrorPolicy } from '../../../generated/mst-gql/uniswap-v3'
+
+export const QueryStoreCreator = RootStoreBase.actions((self) => ({}))
+
+export class CoreGraphStore {
+  queryStore: typeof QueryStoreCreator.Type
+  constructor(private root: RootStore) {
+    this.queryStore = QueryStoreCreator.create(undefined, {
+      gqlHttpClient: createHttpClient(GRAPH_ENDPOINTS.core.arbitrumOne as string),
+      ssr: true,
+    })
+
+    makeAutoObservable(this)
+  }
+
+  /**
+   * When any of the filters change, the query is replaced with new query variables.
+   */
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  get transactionsQuery() {
```

Comment:
```suggestion
  get transactionsQuery(): ReturnType<typeof QueryStoreCreator.Type['queryTransactions']> {
```

This is not super readable either so feel free to disregard, but I like it better than a rule disable.

Not a blocker.

Reply:
ops I meant to merge your PR for "no-return", added this comment only to bypass build for now, removing now

Reply:
I don't think my PR applies here though

Reply:
yea you're right, thought that rule you removed would apply here, added return type here with the actual `Query<{transactions: TransactionModelType}>` since `ReturnType<typeof theFunction>` defeats the purpose of `explicit function return type`, which is to restrict what we return in a function

Reply:
> `ReturnType<typeof theFunction>` defeats the purpose of `explicit function return type`, which is to restrict what we return in a function

True that! Just was suggesting a dirty way of getting rid of the warning